src/lib/gundb/UserStorage.js

```diff
@@ -239,14 +239,31 @@ export class UserStorage {
     return feedItem
   }
 
-  async getAllFeed() {
-    const total = Object.values((await this.feed.get('index')) || {}).reduce((acc, curr) => acc + curr, 0)
+  /**
+   * Returns the amount specified of feeds in desc order (from newest to oldest)
+   * @param {number} amount
+   * @returns {Promise<Array<FeedEvent>>}
+   */
+  async getFeeds(amount: number = 0): Promise<Array<FeedEvent>> {
```

Comment:
You made this for savelastblockprocessed, why not just read the last event of the feed?
further more, you call savelastblockprocessed from updatefeedevent, so basically you just re-read the new event added in updatefeedevent by calling savelastblockprocessed.
Doesn't the event given to updatefeedevent already have the receipt when it has one? I think the whole process can be simplified to checking if the event in  updatefeedevent has a receipt and if the blocknumber in the receipt if higher than the current lastblock saved and then update it.

Reply:
Oh, that was enlightening. Thanks for the clarification!